keras_cv/models/backbones/mobilenet_v3/mobilenet_v3_backbone_test.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import tensorflow as tf
+from absl.testing import parameterized
+from tensorflow import keras
+
+from keras_cv.models.backbones.mobilenet_v3.mobilenet_v3_backbone import (
+    MobileNetV3SmallBackbone,
+)
+from keras_cv.utils.train import get_feature_extractor
+
+
+class MobileNetV3BackboneTest(tf.test.TestCase, parameterized.TestCase):
+    def setUp(self):
+        self.input_batch = tf.ones(shape=(2, 224, 224, 3))
+
+    def test_valid_call(self):
+        model = MobileNetV3SmallBackbone(
+            include_rescaling=False,
+        )
+        model(self.input_batch)
+
+    def test_valid_call_with_rescaling(self):
+        model = MobileNetV3SmallBackbone(
+            include_rescaling=True,
+        )
+        model(self.input_batch)
+
+    @parameterized.named_parameters(
+        ("tf_format", "tf", "model"),
+        ("keras_format", "keras_v3", "model.keras"),
+    )
+    def test_saved_model(self, save_format, filename):
+        model = MobileNetV3SmallBackbone()
+        model_output = model(self.input_batch)
+        save_path = os.path.join(self.get_temp_dir(), filename)
+        model.save(save_path, save_format=save_format)
+        restored_model = keras.models.load_model(save_path)
+
+        # Check we got the real object back.
+        self.assertIsInstance(restored_model, MobileNetV3SmallBackbone)
```

Comment:
Should be

```python
self.assertIsInstance(restored_model, MobileNetV3Backbone)
```

Long story!

Reply:
Right, because we are testing whether the loaded model is an instance of the `MobileNetV3Backbone` class or not to ensure that the loaded model has the same architecture as the original model. Using `MobileNetV3SmallBackbone` here makes no sense.